### ISSUE TITLE
Divided Menu Accessibilty Issue

### ISSUE THREAD

**Issue by mhemesath** [cerner/terra-core#1066]
# Issue Description
Menu components with dividers fail accessibility because the `<hr/>` dividers cannot be direct descendants of `<ul>`.

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
Passes accessibility tests

## Current Behavior
```
List element has direct children that are not allowed inside <li> elements
```

## Steps to Reproduce

1. Install [Axe Chrome Extension](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)
2. Navigate to a [Large Menu Test](http://engineering.cerner.com/terra-core/#/tests/menu-tests/large) and open the menu by clicking on it.
3. Run axe by going to developer tools and clicking axe tab.

## Environment

* Component Version:
* Browser Name and Version:
* Operating System and version (desktop or mobile):

**Comment by mhemesath** [comment]
I'm wondering if this should be solved by making each divided menu be its own list group. That list group can then have the </hr> as a sibling.
